import-export-cli/cmd/aws/init.go

```diff
@@ -0,0 +1,512 @@
+		os.RemoveAll(tmpDir)
+		os.Exit(1)
+		return err
+	}
+	return nil
+}
+
+// loadDefaultAWSDocFromDisk loads document.yaml stored in HOME/.wso2apictl/document.yaml
+func loadDefaultAWSDoc() (*v2.Document, error) {
+	docData, ok := box.Get("/init/document.yaml")
+	if !ok {
+		return nil, errors.New("Error while retrieving document.yaml")
+	}
+	awsDoc := &v2.Document{}
+	err = yaml.Unmarshal(docData, &awsDoc)
+	if err != nil {
+		return nil, err
+	}
+	return awsDoc, nil
+}
+
+func createAWSDocDirectory(docName string) error {
+	awsDocDirectoryPath := initCmdOutputDir + string(os.PathSeparator) + "Docs"
+	dirPath := filepath.Join(awsDocDirectoryPath, filepath.FromSlash(docName))
+	utils.Logln(utils.LogPrefixInfo + "Creating directory " + dirPath)
+	err := os.MkdirAll(dirPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//write document.yaml file
+func writeDocumentFile(docName string, summary string) error {
+	document, err := loadDefaultAWSDoc()
+	docData := &document.Data
+	docData.Name = docName
+	docData.Summary = summary
+	docDataByte, err := yaml2.Marshal(document)
+	if err != nil {
+		return err
+	}
+	apiDocFilePath := filepath.Join(initCmdOutputDir, filepath.FromSlash("Docs" + string(os.PathSeparator) + docName + string(os.PathSeparator) + "document.yaml"))
+	utils.Logln(utils.LogPrefixInfo + "Writing " + apiDocFilePath)
+	err = ioutil.WriteFile(apiDocFilePath, docDataByte, os.ModePerm)
+	return nil
+}
+
+// write AWS API security documents based on APIs security schemes
+func writeAWSSecurityDocs(oas3ByteValue []byte) error {
+	securitySchemes := &v2.SecuritySchemes{}
+	json.Unmarshal(oas3ByteValue, &securitySchemes)
+	schemes := securitySchemes.Components.SecuritySchemes
+	if securitySchemes.ResourcePolicy.Version != "" {
+		docName := "Resource Policy"
+		summary := "This document contains details related to AWS resource policies"
+		err = createAWSDocDirectory(docName)
+		resourcePolicyDocPath := filepath.Join(initCmdOutputDir, filepath.FromSlash("Docs" + string(os.PathSeparator) + docName + string(os.PathSeparator) + docName))
+		utils.Logln(utils.LogPrefixInfo + "Writing " + resourcePolicyDocPath)
+		resourcePolicyDoc, _ := box.Get("/init/resource_policy_doc")
+		err = ioutil.WriteFile(resourcePolicyDocPath, resourcePolicyDoc, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		err = writeDocumentFile(docName, summary)
+		if err != nil {
+			return err
+		}
+	}
+	if schemes.CognitoAuthorizer.AuthType == "cognito_user_pools" {
+		docName := "Cognito Userpool"
+		summary := "This document contains details related to AWS cognito user pools"
+		err = createAWSDocDirectory(docName)
+		cognitoUpDocPath := filepath.Join(initCmdOutputDir, filepath.FromSlash("Docs" + string(os.PathSeparator) + docName + string(os.PathSeparator) + docName))
+		utils.Logln(utils.LogPrefixInfo + "Writing " + cognitoUpDocPath)
+		cognitoUpDoc, _ := box.Get("/init/cognito_userpool_doc")
```

Comment:
Shall we add this path as a constant as I mentioned above?

Reply:
ack